File: sentimental-credit/credit.py
from cs50 import get_int

num = str(get_int("Number: "))

sum = 0
# start, stop, step (inclusive start, exclusive stop)
for i in range(len(num) - 1, -1, -1):
    index = len(num) - i
    # If even (starting in the end)
    if index % 2 == 0:
        n = int(num[i]) * 2
        sum = sum + n % 10 + n // 10
    # Else odd
    else:
        sum = sum + int(num[i])

# Checksum
if (sum % 10 != 0):
    print("INVALID")
# AMEX has 15 digits, starts with 34 and 37
elif (len(num) == 15 and (int(num) // 10**13 in [34, 37])):
    print("AMEX")
# VISA has 13 or 16 digits, starts with 4
elif (len(num) in [13, 16] and (int(num) // 10**12 == 4 or int(num) // 10**15 == 4)):
    print("VISA")
# MASTERCARD has 16 digits, starts with 51 to 55
elif (len(num) == 16 and int(num) // 10**14 in range(51, 56)):
    print("MASTERCARD")
else:
    print("INVALID")

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long number = get_long("Number: ");
    int l = 0;
    int sum = 0;
    int digit = 0;
    int twodigits = 0;

    // Loop
    while (number != 0)
    {
        l = l + 1;            // Number length
        digit = number % 10;  // Digit
        number = number / 10; // Current number

        if (number > 9 && number < 100) // 10 to 99
        {
            twodigits = number;
        }

        // printf("Digit: %i \n", digit);
        // printf("Current Number: %li \n", number);

        // Adding checksum
        if (l % 2 == 0)
        {
            int digit_X2 = digit * 2;
            if (digit_X2 > 9)
            {
                int d1 = digit_X2 % 10;        // 12 % 10 = 2
                int d2 = (digit_X2 / 10) % 10; // 1 % 10 = 1
                sum = sum + d1 + d2;
            }
            else
                sum = sum + digit_X2;
        }
        else
            sum = sum + digit;
    }

    if (sum % 10 != 0)
    {
        printf("INVALID\n");
    }
    else if (l == 15 && (twodigits == 34 || twodigits == 37))
    {
        printf("AMEX\n");
    }
    else if ((l == 13 || l == 16) && digit == 4)
    {
        printf("VISA\n");
    }
    else if (l == 16 && (twodigits == 51 || twodigits == 52 || twodigits == 53 || twodigits == 54 ||
                         twodigits == 55))
    {
        printf("MASTERCARD\n");
    }
    else
    {
        printf("INVALID\n");
    }
}
